File: Labs/Lab 5/TreeNode.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

    def isValidBST(self, root):
        if root is None:
            return True
        if root.left and root.left.val >= root.val:
            return False
        if root.right and root.right.val <= root.val:
            return False
        return self.isValidBST(root.left) and self.isValidBST(root.right)

File: HW/HW 10/TreeNode.java
import java.util.*;
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
    public List<List<Integer>> levelOrder(TreeNode root)
    {
        List<List<Integer>> complete = new ArrayList<>();
        if (root == null)
        {
            return complete;
        }

        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);

        while (!queue.isEmpty())
        {
            int level = queue.size();
            List<Integer> currentLevel = new ArrayList<>();

            for (int i = 0; i < level; i++)
            {
                TreeNode node = queue.poll();
                currentLevel.add(node.val);
                if (node.left != null)
                {
                    queue.offer(node.left);
                }
                if (node.right != null)
                {
                    queue.offer(node.right);
                }
            }
            complete.add(currentLevel);
        }
        return complete;

    }

}
